examples/litestar_app.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from abc import abstractmethod
+from typing import Protocol, Annotated
+
+import uvicorn
+from litestar import Controller, get, Litestar
+
+from dishka import Provider, Scope, provide
+from dishka.integrations.base import Depends
+from dishka.integrations.litestar import (
+    inject,
+    setup_dishka,
+    startup_dishka,
+    make_dishka_container,
+    shutdown_dishka
+)
+
+
+# app core
+class DbGateway(Protocol):
+    @abstractmethod
+    def get(self) -> str:
+        raise NotImplementedError
+
+
+class FakeDbGateway(DbGateway):
+    def get(self) -> str:
+        return "Hello"
+
+
+class Interactor:
+    def __init__(self, db: DbGateway):
+        self.db = db
+
+    def __call__(self) -> str:
+        return self.db.get()
+
+
+# app dependency logic
+class AdaptersProvider(Provider):
+    @provide(scope=Scope.REQUEST)
+    def get_db(self) -> DbGateway:
+        return FakeDbGateway()
+
+
+class InteractorProvider(Provider):
+    i1 = provide(Interactor, scope=Scope.REQUEST)
+
+
+class MainController(Controller):
+    path = '/'
+
+    @get()
+    @inject
+    async def index(self, *, interactor: Annotated[Interactor, Depends()]) -> str:
+        result = interactor()
+        return result
+
+
+def create_app():
+    logging.basicConfig(
+        level=logging.WARNING,
+        format='%(asctime)s  %(process)-7s %(module)-20s %(message)s',
+    )
+    app = Litestar(
+        route_handlers=[MainController],
+        on_startup=[startup_dishka],
```

Comment:
Can we reuse slightly modified asgi middleware from fastapi?

Reply:
You mean something like this?
```
def add_request_container_middleware(app: ASGIApp):
    async def middleware(scope: Scope, receive: Receive, send: Send):
        request = Request(scope)
        async with request.app.state.dishka_container({Request: request}) as request_container:
            request.state.dishka_container = request_container
            await app(scope, receive, send)

    return middleware
```

Reply:
For fastapi I have 2 middlewares:

1. ASGI - it controls Scope.APP. We can try to unify for all asgi apps. https://github.com/reagento/dishka/blob/develop/src/dishka/integrations/fastapi.py#L47
2. Framework HTTP - it  controls Scope.REQUEST. It is not asgi middleware because we want to access framework specific Request object.  https://github.com/reagento/dishka/blob/develop/src/dishka/integrations/fastapi.py#L39